libs/authentication/api/src/lib/authentication.guard.ts

```diff
@@ -0,0 +1,10 @@
+import {AuthGuard} from '@nestjs/passport'
+import {ExecutionContext} from '@nestjs/common'
+import {GqlExecutionContext} from '@nestjs/graphql'
+
+export class AuthenticationGuard extends AuthGuard('session') {
+  public getRequest(context: ExecutionContext) {
+    const ctx = GqlExecutionContext.create(context)
+    return ctx.getContext().req
+  }
+}
```

Comment:
I understand this guard now checks if user has session and if it's valid, right?
If it is it's attaching user to context, right?

Reply:
Attaching the user is already done by the optional authentication guard, but this will also do it. 
This is for public routes where we want the user to be authenticated but doesn't need any specific roles